File: src/Application Framework/System/SystemData.cs
﻿using System.Data;
using Microsoft.VisualBasic;

namespace FSM.Entity
{
    namespace Sys
    {
        public class SystemData
        {
            private const Enums.DatabaseSystems _DataBaseServerType = Enums.DatabaseSystems.Microsoft_SQL_Server;

            public Enums.DatabaseSystems DataBaseServerType
            {
                get
                {
                    return _DataBaseServerType;
                }
            }

            private Configuration _Configuration = new Configuration();

            public Configuration Configuration
            {
                get
                {
                    return _Configuration;
                }
            }

            private const string _DeletedPostfix = " (Deleted)";

            public string DeletedPostfix
            {
                get
                {
                    return _DeletedPostfix;
                }
            }

            private const string _Title = "Field Service Manager - Beta";

            public string Title
            {
                get
                {
                    return _Title;
                }
            }

            private const string _Description = "Field Management System";

            public string Description
            {
                get
                {
                    return _Description;
                }
            }

            private const string _Company = "Gasway Services Ltd";

            public string Company
            {
                get
                {
                    return _Company;
                }
            }

            private const string _Product = "FSM";

            public string Product
            {
                get
                {
                    return _Product;
                }
            }

            private const string _Copyright = "FSM";

            public string Copyright
            {
                get
                {
                    return _Copyright;
                }
            }

            private const string _Trademark = "FSM";

            public string Trademark
            {
                get
                {
                    return _Trademark;
                }
            }

            public DataTable DLLInformation
            {
                get
                {
                    var dt = new DataTable();
                    dt.Columns.Add("Name");
                    dt.Columns.Add("Version");
                    foreach (System.Reflection.AssemblyName assem in GetType().Assembly.GetReferencedAssemblies())
                        dt.Rows.Add(new string[] { assem.Name, assem.Version.ToString() });
                    dt.TableName = Enums.TableNames.NOT_IN_DATABASE_DLLs.ToString();
                    return dt;
                }
            }

            private const string _Address = "Exel Computer Systems plc" + Constants.vbCrLf + Constants.vbCrLf + "Bothe Hall" + Constants.vbCrLf + "Sawley" + Constants.vbCrLf + "Nottingham" + Constants.vbCrLf + "NG10 3XL";

            public string Address
            {
                get
                {
                    return _Address;
                }
            }

            private const string _Telephone = "0115 946 0101";

            public string Telephone
            {
                get
                {
                    return _Telephone;
                }
            }

            private const string _Fax = "0115 946 0101\"";

            public string Fax
            {
                get
                {
                    return _Fax;
                }
            }

            public string ContactUs
            {
                get
                {
                    return Address + Constants.vbCrLf + Constants.vbCrLf + Constants.vbCrLf + "Tel : " + Telephone + Constants.vbCrLf + Constants.vbCrLf + Constants.vbCrLf;
                }
            }
        }
    }
}

File: src/Application Framework/System/SystemData.vb
Namespace Entity

    Namespace Sys

        Public Class SystemData

            Private Const _DataBaseServerType As Enums.DatabaseSystems = Enums.DatabaseSystems.Microsoft_SQL_Server

            Public ReadOnly Property DataBaseServerType() As Enums.DatabaseSystems
                Get
                    Return _DataBaseServerType
                End Get
            End Property

            Private _Configuration As New Configuration

            Public ReadOnly Property Configuration() As Configuration
                Get
                    Return _Configuration
                End Get
            End Property

            Private Const _DeletedPostfix As String = " (Deleted)"

            Public ReadOnly Property DeletedPostfix() As String
                Get
                    Return _DeletedPostfix
                End Get
            End Property

            Private Const _Title As String = "Field Service Manager - Beta"

            Public ReadOnly Property Title() As String
                Get
                    Return _Title
                End Get
            End Property

            Private Const _Description As String = "Field Management System"

            Public ReadOnly Property Description() As String
                Get
                    Return _Description
                End Get
            End Property

            Private Const _Company As String = "Gasway Services Ltd"

            Public ReadOnly Property Company() As String
                Get
                    Return _Company
                End Get
            End Property

            Private Const _Product As String = "FSM"

            Public ReadOnly Property Product() As String
                Get
                    Return _Product
                End Get
            End Property

            Private Const _Copyright As String = "FSM"

            Public ReadOnly Property Copyright() As String
                Get
                    Return _Copyright
                End Get
            End Property

            Private Const _Trademark As String = "FSM"

            Public ReadOnly Property Trademark() As String
                Get
                    Return _Trademark
                End Get
            End Property

            Public ReadOnly Property DLLInformation() As DataTable
                Get
                    Dim dt As New DataTable
                    dt.Columns.Add("Name")
                    dt.Columns.Add("Version")
                    For Each assem As System.Reflection.AssemblyName In Me.GetType().Assembly.GetReferencedAssemblies
                        dt.Rows.Add(New String() {assem.Name, assem.Version.ToString()})
                    Next
                    dt.TableName = Enums.TableNames.NOT_IN_DATABASE_DLLs.ToString
                    Return dt
                End Get
            End Property

            Private Const _Address As String = "Exel Computer Systems plc" & vbCrLf & vbCrLf & "Bothe Hall" & vbCrLf & "Sawley" & vbCrLf & "Nottingham" & vbCrLf & "NG10 3XL"

            Public ReadOnly Property Address() As String
                Get
                    Return _Address
                End Get
            End Property

            Private Const _Telephone As String = "0115 946 0101"

            Public ReadOnly Property Telephone() As String
                Get
                    Return _Telephone
                End Get
            End Property

            Private Const _Fax As String = "0115 946 0101"""

            Public ReadOnly Property Fax() As String
                Get
                    Return _Fax
                End Get
            End Property

            Public ReadOnly Property ContactUs() As String
                Get
                    Return Address & vbCrLf & vbCrLf & vbCrLf & "Tel : " & Telephone & vbCrLf & vbCrLf & vbCrLf
                End Get
            End Property

        End Class

    End Namespace

End Namespace